flaky test(legacylibrarian): TestBuildGoLibraries

Test failed at unrelated change (https://github.com/googleapis/librarian/pull/3856). Passed with rerun.

full log: https://github.com/googleapis/librarian/actions/runs/21680243528/job/62512278203
```
--- FAIL: TestBuildGoLibraries (0.00s)
    --- FAIL: TestBuildGoLibraries/go_libraries (0.00s)
        legacylibrarian_test.go:376: mismatch (-want +got):
              []*config.Library{
              	&{
            - 		Name:                "another-library",
            + 		Name:                "example-library",
              		Version:             "",
            - 		APIs:                []*config.API{&{Path: "google/another/api/v1"}},
            + 		APIs:                []*config.API{&{Path: "google/example/api/v1"}},
              		CopyrightYear:       "",
              		DescriptionOverride: "",
              		... // 9 identical fields
              		Veneer: false,
              		Dart:   nil,
            - 		Go:     nil,
            + 		Go: &config.GoModule{
            + 			DeleteGenerationOutputPaths: []string{"internal/generated/snippets/storage/internal"},
            + 			GoAPIs: []*config.GoAPI{
            + 				&{
            + 					Path:            "google/maps/fleetengine/v1",
            + 					ClientDirectory: "proto_package: maps.fleetengine.v1",
            + 					DisableGAPIC:    true,
            + 					NestedProtos:    []string{...},
            + 					...
            + 				},
            + 			},
            + 			ModulePathVersion: "v2",
            + 		},
              		Python: nil,
              		Rust:   nil,
              	},
              	&{
            - 		Name:                "example-library",
            + 		Name:                "another-library",
              		Version:             "",
            - 		APIs:                []*config.API{&{Path: "google/example/api/v1"}},
            + 		APIs:                []*config.API{&{Path: "google/another/api/v1"}},
              		CopyrightYear:       "",
              		DescriptionOverride: "",
              		... // 9 identical fields
              		Veneer: false,
              		Dart:   nil,
            - 		Go: &config.GoModule{
            - 			DeleteGenerationOutputPaths: []string{"internal/generated/snippets/storage/internal"},
            - 			GoAPIs: []*config.GoAPI{
            - 				&{
            - 					Path:            "google/maps/fleetengine/v1",
            - 					ClientDirectory: "proto_package: maps.fleetengine.v1",
            - 					DisableGAPIC:    true,
            - 					NestedProtos:    []string{...},
            - 					...
            - 				},
            - 			},
            - 			ModulePathVersion: "v2",
            - 		},
            + 		Go:     nil,
              		Python: nil,
              		Rust:   nil,
              	},
              }
```

--- a/tool/cmd/migrate/legacylibrarian_test.go
+++ b/tool/cmd/migrate/legacylibrarian_test.go
@@ -21,6 +21,7 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/googleapis/librarian/internal/config"
 	"github.com/googleapis/librarian/internal/legacylibrarian/legacyconfig"
 )
@@ -372,7 +373,9 @@ func TestBuildGoLibraries(t *testing.T) {
 	} {
 		t.Run(test.name, func(t *testing.T) {
 			got := buildGoLibraries(test.input)
-			if diff := cmp.Diff(test.want, got); diff != "" {
+			if diff := cmp.Diff(test.want, got, cmpopts.SortSlices(func(a, b *config.Library) bool {
+				return a.Name < b.Name
+			})); diff != "" {
 				t.Errorf("mismatch (-want +got):\n%s", diff)
 			}
 		})

